File: programmers/level1/64061.py
'''
크레인 인형뽑기 게임
'''


def solution(board, moves):

    arry = []
    for i in moves:
      for j in range(len(board)):
        if board[j][i-1] != 0:
          arry.append(board[j][i-1])
          board[j][i-1] = 0
          break

    r = 0
    n = 1
    while n < len(arry):
        if arry[n] == arry[n-1]:
          r += 2
          del arry[n]
          del arry[n-1]
          if n != 1:
            n -= 1
        else:  
          n += 1  
    return r


'''
stack을 사용해서 푸는 방법
'''
def solution(board, moves):
    stacklist = []
    answer = 0

    for i in moves:
        for j in range(len(board)):
            if board[j][i-1] != 0:
                stacklist.append(board[j][i-1])
                board[j][i-1] = 0

                if len(stacklist) > 1:
                    if stacklist[-1] == stacklist[-2]:
                        stacklist.pop(-1)
                        stacklist.pop(-1)
                        answer += 2     
                break

    return answer

File: programmers/level1/64061.js
//크레인 인형뽑기 게임

function solution(b, m) {
    
    let stacklist = [];
    let answer = 0;
    const bl = b.length;
    const ml = m.length;
    
    for (var i = 0; i < ml; i++){
        for (var j = 0; j < bl; j++){
            if (b[j][m[i]-1] !== 0){
                stacklist.push(b[j][m[i]-1])
                b[j][m[i]-1] = 0
                if (stacklist.length > 1){
                    if (stacklist[stacklist.length-1] === stacklist[[stacklist.length-2]]){
                        stacklist.pop()
                        stacklist.pop()
                        answer += 2 
                    }
                }
                break
            }
        }
    };

    return answer
}
